lix/source-code/client/src/git-http/client.js

```diff
@@ -130,35 +130,81 @@ function fromStream(stream) {
 /* eslint-env browser */
 
 /**
- * HttpClient
+ * MakeHttpClient
  *
- * @param {GitHttpRequest} request
- * @returns {Promise<GitHttpResponse>}
+ * @param { verbose?: boolean, desciption?: string, onReq: ({body: any, url: string }) => {body: any, url: string} }
+ * @returns HttpClient
  */
-async function request({ url, method = "GET", headers = {}, body }) {
-	// onProgress param not used
-	// streaming uploads aren't possible yet in the browser
-	if (body) {
-		body = await collect(body)
-	}
-	const res = await fetch(url, { method, headers, body, credentials: "include" })
-	const iter =
-		res.body && res.body.getReader
-			? fromStream(res.body)
-			: [new Uint8Array(await res.arrayBuffer())]
-	// convert Header object to ordinary JSON
-	headers = {}
-	for (const [key, value] of res.headers.entries()) {
-		headers[key] = value
-	}
-	return {
-		url: res.url,
-		method: res.method,
-		statusCode: res.status,
-		statusMessage: res.statusText,
-		body: iter,
-		headers: headers,
+export function makeHttpClient({ verbose, description, onReq, onRes }) {
```

Comment:
@martin-lysk  i assumed we only would need one at a time, no-progress seems so small of a change it could be included in the other rewrite functions, but if its much simpler to have multiple features, we can make them either chainable so we can do noProgress(replaceWant) or combine them into a not single purpose function that gets params similar like how you do it at the moment.